File: sdk/web/app/lib/validation.ts
import { COPY } from "../copy.ts";

const ACCOUNT = /^[A-Za-z0-9_]{3,32}$/;

export function validateAccount(raw: string): string | undefined {
  if (!ACCOUNT.test(raw.trim())) {
    return COPY.invalidAccount;
  }
  return undefined;
}

export function validatePassword(raw: string): string | undefined {
  if (raw.length < 8 || raw.length > 128) {
    return COPY.invalidPassword;
  }
  return undefined;
}

export function validateConfirm(password: string, confirm: string): string | undefined {
  if (password !== confirm) {
    return COPY.mismatch;
  }
  return undefined;
}

File: sdk/mobile/lib/core/validation.dart
library;

import '../copy.dart';

final _account = RegExp(r'^[A-Za-z0-9_]{3,32}$');

String? validateAccount(String raw) {
  if (!_account.hasMatch(raw.trim())) {
    return Copy.invalidAccount;
  }
  return null;
}

String? validatePassword(String raw) {
  if (raw.length < 8 || raw.length > 128) {
    return Copy.invalidPassword;
  }
  return null;
}

String? validateConfirm(String password, String confirm) {
  if (password != confirm) {
    return Copy.mismatch;
  }
  return null;
}
